plugins/mac-videotoolbox/encoder.c

```diff
@@ -497,6 +494,14 @@ static void update_params(struct vt_encoder *enc, obs_data_t *settings)
 	// also sets the enc->vt_pix_fmt
 	vt_video_info(enc, &info);
 
+	if (info.format != voi->format) {
```

Comment:
In the end, this does what it is supposed to, but it seems unfortunate that we pass an "invalid" setting into `vt_video_info`, which then also sets the fallback value of NV12, only to then compare the fallback value with the value in OBS settings.

Ideally, `vt_video_info` (which is also unfortunately named as it doesn't indicate a manipulation of data) should return a boolean when it was given an invalid colour format.

Reply:
> Ideally, vt_video_info (which is also unfortunately named as it doesn't indicate a manipulation of data) should return a boolean when it was given an invalid colour format.

I was thinking that as well orgininally, unfortunately it's used directly in the encoder info struct `.get_video_info`. I could refactor the code a bit though.

Reply:
Cleaned up the existing code.
Removed `.get_video_info` from the struct, since that "can be used to tell the back-end that the video data needs to be automatically converted to a different video format or specific size before being sent to the encoder" (docs), which is exactly the kind of fallback we don't want.